tests/models/whisper/test_modeling_whisper.py

```diff
@@ -3562,6 +3687,37 @@ def test_decoder_model_attn_mask_past(self):
             config=config, input_ids=inputs_dict["input_ids"]
         )
 
+    @unittest.skip("TODO Sanchit: fix failing test")
```

Comment:
This one's ready for a second review @gante. The major changes from the previous review include:
1. Adding the `EncoderDecoderCache` abstraction (lightweight wrapper around cache classes), including the methods required for compatibility with sequential beam search
2. Adding a boolean flag to determine whether we're in decoding step 1 or >1

TODOs:
- [ ] Fix last failing test

Reply:
What's the difference between this and `test_custom_4d_attention_mask` above?

Reply:
The one here is for the standalone decoder model (`WhisperForCausalLM`). In practice, this class is only used in the special case of speculative decoding where the main model encoder matches the assistant model encoder, meaning we only load the assistant model **decoder** in addition to the main model encoder-decoder.

The test above is for the generic encoder-decoder model (`WhisperForConditionalGeneration`)